swift/ql/test/library-tests/dataflow/dataflow/test.swift

```diff
@@ -631,3 +631,30 @@ func testSwap() {
     sink(arg: x) // $ SPURIOUS: flow=628
     sink(arg: y) // $ flow=628
 }
+
+func testArray() {
+    var arr1 = [1,2,3]
+    sink(arg: arr1[0])
+    arr1[1] = source()
+    sink(arg: arr1[0]) // $ flow=638
+    sink(arg: arr1)
+
+    var arr2 = [source()]
+    sink(arg: arr2[0]) // $ flow=642
+
+    var matrix = [[source()]]
+    sink(arg: matrix[0])
+    sink(arg: matrix[0][0]) // $ flow=645
+
+    var arr3 = [1]
+    var arr4 = arr2 + arr3
+    sink(arg: arr3[0])
+    sink(arg: arr4[0]) // $ MISSING: flow=642
+
+    var arr5 = Array(repeating: source(), count: 2)
+    sink(arg: arr5[0]) // $ MISSING: flow=654
```

Comment:
Again it's possible a model of the initializer is all we need here.  It can wait as follow-up.

Reply:
Agreed.